core/src/main/scala/org/apache/spark/ui/jobs/JobProgressListener.scala

```diff
@@ -528,7 +528,13 @@ class JobProgressListener(conf: SparkConf) extends SparkListener with Logging {
         new StageUIData
       })
       val taskData = stageData.taskData.get(taskId)
-      val metrics = TaskMetrics.fromAccumulatorInfos(accumUpdates)
+      val accumsFiltered = if (conf.get(TASK_METRICS_TRACK_UPDATED_BLOCK_STATUSES)) {
+        accumUpdates
+      } else {
+        accumUpdates.filter(info => info.name.isDefined && info.update.isDefined && info.name !=
```

Comment:
How about I just revert this. I'm not sure its worth an assert here.  the updated block statuses are just going to be empty so setting it to empty isn't going to hurt anything.